### PR TITLE
ANW-97: Inactive users

### PR DESCRIPTION
**Pull request by avatar382**
## Description
Allows users to be marked inactive

## Related JIRA Ticket or GitHub Issue
https://archivesspace.atlassian.net/jira/software/c/projects/ANW/issues/ANW-97

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have authority to submit this code.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: backend/app/controllers/users.rb [hunk 1]
```diff
@@ -41,12 +42,16 @@ class ArchivesSpaceService < Sinatra::Base
...
+    .params([ "show_inactive", BooleanParam, "Flag to show inactive users", :optional => true ])
     .paginated(true)
     .permissions([])
     .returns([200, "[(:resource)]"]) \
   do
-    handle_listing(User, params, {:exclude => {:id => User.unlisted_user_ids}})
+    if params[:show_inactive]
+      handle_listing(User, params, {:exclude => {:id => User.unlisted_user_ids}})
+    else
+      handle_listing(User, params, {:exclude => {:id => User.unlisted_user_ids, :is_active_user => 0}})
```
Comment (quoideneuf) [inline backend/app/controllers/users.rb:53]:
```suggestion
      handle_listing(User, params, {:exclude => {:id => User.unlisted_user_ids}, :is_active_user => 1})
```
I think sending two keys to exclude will only exclude records matching both conditions, so you have to stick to one exclude field (or modify crud_helpers to support exclude chaining) and use a filter for `:is_active_user`

File: frontend/app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+                    "manage_repository" => [:manage_access, :edit_groups, :update_groups, :complete],
+                    :public => [:new, :create]
 
   before_action :account_self_service, :only => [:new, :create]
   before_action :user_needs_to_be_a_user_manager_or_new_user, :only => [:new, :create]
   before_action :user_needs_to_be_a_user, :only => [:show]
 
-
   def index
+    show_inactive = params[:show_inactive] || false
```
Comment (quoideneuf) [inline frontend/app/controllers/users_controller.rb:12]: what is this line doing? it doesn't appear that `show_inactive` is used.

**Review by quoideneuf — CHANGES_REQUESTED** [review]
It looks like the `show_inactive` param in both controllers is not doing anything. Maybe it can just be removed?

**Comment by avatar382** [comment]
> It looks like the `show_inactive` param in both controllers is not doing anything. Maybe it can just be removed?

Good catch, since all the users is shown in one view, you're right -- no need for the param. I've removed it from both controllers.

**Review by quoideneuf — APPROVED** [review]

**Pull request merged**